package/MDAnalysis/analysis/align.py

```diff
@@ -359,27 +415,141 @@ def alignto(mobile, reference, select="all", mass_weighted=False,
+            Boolean, if true will rmsd will be mass-weighted corresponding to
+            the atoms selected in the reference trajectory
+        tol_mass : float, optional
+            Tolerance given to `get_matching_atoms` to find appropriate atoms
+        strict : boolean, optional
+            Force `get_matching_atoms` to fail if atoms can't be found using exact
+            methods
+        force : boolean, optional
+            Force overwrite of filename for rmsd-fitting
+        quiet : boolean, optional
+            Set logger to show minimal information
+        start : int, optional
+            First frame of trajectory to analyse, Default: 0
+        stop : int, optional
+            Last frame of trajectory to analyse, Default: -1
+        step : int, optional
+            Step between frames to analyse, Default: 1
+
+        """
+        self.quiet = quiet
+        if self.quiet:
+            # should be part of a try ... finally to guarantee restoring the log level
+            logging.disable(logging.WARN)
+
+        self.mobile = mobile
+        self.reference = reference
+        frames = self.mobile.trajectory
+
+        select = rms.process_selection(select)
+        self.ref_atoms = self.reference.select_atoms(*select['reference'])
+        self.mobile_atoms = self.mobile.select_atoms(*select['mobile'])
+        self.natoms = self.mobile_atoms.n_atoms
+
+        kwargs.setdefault('remarks', 'RMS fitted trajectory to reference')
+
+        if filename is None:
+            path, fn = os.path.split(frames.filename)
+            filename = os.path.join(path, prefix + fn)
+            logger.info('none filename: {0}'.format(filename))
+
+        if os.path.exists(filename) and not force:
+            raise IOError('Filename already exists in path and force is not set to True')
+        self.filename = filename
+
+        self.natoms = self.mobile_atoms.n_atoms
+        self.ref_atoms, self.mobile_atoms = get_matching_atoms(self.ref_atoms,
+                            self.mobile_atoms, tol_mass=tol_mass, strict=strict)
+
+        self.writer = mda.Writer(filename, self.natoms)
+
+        if mass_weighted:
+            # if performing a mass-weighted alignment/rmsd calculation
+            self.weights = self.ref_atoms.masses
+        else:
+            self.weights = None
+
+        logger.info("RMS-fitting on {0:d} atoms.".format(len(self.ref_atoms)))
+        self._setup_frames(frames, start,stop,step)
+
+    def _prepare(self):
+        # reference centre of mass system
+        self.ref_com = self.ref_atoms.center_of_mass()
+        self.ref_coordinates = self.ref_atoms.positions - self.ref_com
+        # allocate the array for selection atom coords
+        self.rmsd = np.zeros((self.nframes,))
+
+    def _single_frame(self):
+        index = self._ts.frame
+
+        self.mobile_com = self.mobile_atoms.center_of_mass()
+        self.mobile_coordinates = self.mobile_atoms.positions.copy() - self.mobile_com
```

Comment:
http://shop.oreilly.com/product/0636920028963.do

I don't think I'm really good at benchmarking but I developed some intuition over the time of what is a really stupid thing todo by making all the stupid mistakes. But please remote the explicit copy you have right now. the prealloc can be handled in another PR where you focus on performance.